converter/codec.go

```diff
@@ -172,9 +172,17 @@ func (e *CodecDataConverter) decode(payloads []*commonpb.Payload) ([]*commonpb.P
 // ToPayload implements DataConverter.ToPayload performing encoding on the
 // result of the parent's ToPayload call.
 func (e *CodecDataConverter) ToPayload(value interface{}) (*commonpb.Payload, error) {
```

Comment:
I don't think raw value logic should be in the `CodecDataConverter`, I think `CompositeDataConverter` is enough. Codecs have nothing to do with raw value, raw value is about type conversion, raw value handling should be done by the parent/underlying data converter.

Reply:
I disagree, from the issue `Encoding/decoding of the payload is still done by the system` so the `CodecDataConverter` has to be aware of it

Reply:
You can't assume that the parent data convert will have support for `RawValue`. There is no reason not to build in support for `RawValue` into the `CodecDataConverter`

Reply:
> You can't assume that the parent data convert will have support for RawValue

Right, same as if the codec was not present, but adding new type conversion behavior whether you have the codec on or off is confusing. It's not like, due to the inability to "assume that the parent data convert will have support for RawValue" we're going to everywhere the data converter is using and altering the code based on that assumption, we are just hoping that type conversion properly respects this value.

Reply:
We can clarify in the `CodecDataConverter` docs, but the `CodecDataConverter` should handle `RawValue` absolutely. It is not a responsibility we should push onto our users .

Reply:
It is a responsibility we should push onto users not using a composite payload converter, but ok if you want to _also_ add to codec data converter (it's a bit confusing/inconsistent that only codec users don't have to properly handle this type in non-composite payload converters they write). There are a lot of people not using codecs, so we need to be consistent in our expectation of user responsibility for handling this value regardless of codec.